daemon/endpoint.go

```diff
@@ -57,10 +57,6 @@ func (d *Daemon) linkContainerID(ep *endpoint.Endpoint) {
 
 // insertEndpoint inserts the ep in the endpoints map. To be used with endpointsMU locked.
 func (d *Daemon) insertEndpoint(ep *endpoint.Endpoint) {
-	if ep.Status == nil {
```

Comment:
I think you need to add this `if ep.Status != nil { ep.Status = endpoint.NewEndpointStatus()}` here:
https://github.com/cilium/cilium/blob/master/daemon/state.go#L171